### PR TITLE
Update `Dockerfile`s with small improvements

### LINKED ISSUE #1015: Improvements to `Dockerfile`s
## Problem

The `Dockerfile`s at `docker/local_postgres/Dockerfile` and `docker/airflow/Dockerfile` can be improved.

## Description

- Setting the `DEBIAN_FRONTEND` parameter is [discouraged by Docker](https://docs.docker.com/engine/faq/#why-is-debian_frontendnoninteractive-discouraged-in-dockerfiles) as it is purely a cosmetic change. This line can be dropped.

  https://github.com/WordPress/openverse-catalog/blob/de078ce8ed82df3565db0ff3fe1bc8e369468183/docker/local_postgres/Dockerfile#L6

- The right env variable for preventing `pip` cache is `PIP_NO_CACHE_DIR`.

  https://github.com/WordPress/openverse-catalog/blob/de078ce8ed82df3565db0ff3fe1bc8e369468183/docker/local_postgres/Dockerfile#L7

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #1015 by @dhruvkb

## Description

This PR updates the `Dockerfile`s with small improvements.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docker/airflow/Dockerfile [hunk 1]
```diff
@@ -15,9 +15,8 @@ ENV PYTHONPATH=${DAGS_FOLDER}
 ENV PATH=${AIRFLOW_HOME}/.local/bin:$PATH
 
 # Container optimizations
-ENV PIPNOCACHEDIR=1
-ENV DEBIAN_FRONTEND=noninteractive
 ENV PYTHONUNBUFFERED=1
+ENV PIP_NO_CACHE_DIR=1
```
Comment (AetherUnbound) [inline docker/airflow/Dockerfile:19]: 🤦🏼‍♀️ 😅

File: docker/local_postgres/Dockerfile [hunk 2]
```diff
@@ -3,19 +3,18 @@ FROM postgres:13.6
...
-    libpq-dev \
+RUN apt-get update \
+    && apt-get -yqq install \
+      python3-boto3 \
+      postgresql-plpython3-13 \
+      python3-pip \
+      libpq-dev \
     && apt-get autoremove -y \
     && rm -rf /var/lib/apt/lists/* \
-    && pip3 install -U pip \
```
Comment (AetherUnbound) [inline docker/local_postgres/Dockerfile:18]: Why remove the pip update step here?

**Review by AetherUnbound — APPROVED** [review]
Thanks for this @dhruvkb! One question but nothing to block.

File: docker/local_postgres/Dockerfile (hunk 2 above)
Comment (zackkrida) [reply docker/local_postgres/Dockerfile:18]: It's being combined with the pgcli installation in the following line:

`&& pip3 install -U pip pgcli==${PGCLI_VERSION}`

Not sure about that. Will the -U flag impact pgcli?

Comment (dhruvkb) [reply docker/local_postgres/Dockerfile:18]: I checked, `-U` does not affect dependencies with a version number specified.

Comment (AetherUnbound) [reply docker/local_postgres/Dockerfile:18]: Ahh, my mistake, I missed that!

**Review by zackkrida — APPROVED** [review]

**Pull request merged**